os/android/iahwc2.cpp

```diff
@@ -637,6 +637,14 @@ HWC2::Error IAHWC2::HwcDisplay::PresentDisplay(int32_t *retire_fence) {
       cursor_z_order = l.second.z_order();
       continue;
     }
+#ifndef FORCE_ALL_DEVICE_TYPE
+    if ((l.second.validated_type() == HWC2::Composition::Client) &&
```

Comment:
do you know in what case and why this situation will be met?

Reply:
it is a MACRO added by Dong. Default should be force

Reply:
@dyang23  Please also help to review

Reply:
It is SurfaceFlinger's behavior. Now, we can meet it in Home and App page. I suppose when SF do composition for Client layer, this issue could happen.

Reply:
Till now, this MACRO is not defined. it is not force all layer "Device". I will prepare one patch for it